tests/test_weconnect.py

```diff
@@ -0,0 +1,258 @@
+        self.client().post('/api/v1/auth/register', 
+            data=json.dumps(self.login_user), 
+            content_type='application/json')
+        to_response = self.client().post('/api/v1/auth/login', 
+            data=json.dumps(self.login_user), 
+            headers={"content-type":'application/json'})
+        data = json.loads(to_response.data.decode())
+        self.token = data['token']
+
+    def test_register_user(self):
+        """Test api can register new user"""
+        response = self.client().post('/api/v1/auth/register', 
+            data=json.dumps(self.test1_user), 
+            content_type='application/json')
+        self.assertEqual(response.status_code, 201)
+        self.assertIn('User registered successfully', str(response.data))
+
+    def test_cannot_create_duplicate_user(self):
+        """Test api cannot create duplicate user"""
+        response = self.client().post('/api/v1/auth/register', 
+            data=json.dumps(self.login_user), 
+            content_type='application/json')
+        self.assertIn('Username already exists', str(response.data))
+
+    def test_cannot_create_with_duplicate_email(self):
+        """Test api cannot register user with duplicate email"""
+        response = self.client().post('/api/v1/auth/register', 
+            data=json.dumps(self.email_user), 
+            content_type='application/json')
+        self.assertIn('Email already exists. Try another one', 
+            str(response.data))
+
+    def test_get_users(self):
+        """Test api can get all users"""
+        response = self.client().get('/api/v1/auth/users', 
+            headers={'content-type':'application/json', 
+                'x-access-token':self.token})
+        self.assertEqual(response.status_code, 200)
+
+    def test_login_user(self):
+        """Test api can login registered user"""
+        self.client().post('/api/v1/auth/register', 
+            data=json.dumps(self.test_user), 
+            headers={'content-type':'application/json'})
+        response = self.client().post('/api/v1/auth/login', 
+            data=json.dumps(self.test_user), content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+
+    def test_reset_password(self):
+        """Test api can reset user password"""
+        self.client().post('/api/v1/auth/register', 
+            data=json.dumps(self.login_user), 
+            headers={'content-type':'application/json'})
+        self.client().post('/api/v1/auth/login', 
+            data=json.dumps(self.login_user), content_type='application/json')
+        response = self.client().post('/api/v1/auth/reset-password', 
+            data=json.dumps(self.reset_passw), 
+            headers={'content-type':'application/json', 
+                'x-access-token':self.token})
+        self.assertEqual(response.status_code, 200)
+        # self.assertIn("Password reset successful", str(response.data))
```

Comment:
Remove commented code if not needed.

Reply:
Thanks, @jm-wanja  for the review. Working on it straight away.